pkg/webhook/mutation/pod/oneagent/mutator.go

```diff
@@ -112,6 +112,37 @@ func (mut *Mutator) ensureInitSecret(request *dtwebhook.MutationRequest) error {
 	return nil
 }
 
+func (mut *Mutator) isInjectionPossible(request *dtwebhook.MutationRequest) (bool, string) {
+	reasons := []string{}
+
+	status := request.DynaKube.Status
+
+	_, err := request.DynaKube.TenantUUIDFromConnectionInfoStatus()
+	if err != nil {
+		log.Info("tenant UUID is not available, OneAgent cannot be injected", "pod", request.PodName())
+
+		reasons = append(reasons, EmptyTenantUUIDReason)
+	}
+
+	if !request.DynaKube.IsOneAgentCommunicationRouteClear() {
+		log.Info("OneAgent communication route is not clear, OneAgent cannot be injected", "pod", request.PodName())
+
+		reasons = append(reasons, EmptyConnectionInfoReason)
+	}
+
+	if status.CodeModules.Version == "" {
```

Comment:
What happens in case of `codeModulesImage`?

Reply:
if `codeModulesImage` is configured then the `status.CodeModules.Version` is empty and `status.CodeModules.ImageID` is set

Reply:
also, you could use:
https://github.com/Dynatrace/dynatrace-operator/blob/a69eb0220442afcac5d06fbb9be1eb22ec787486/pkg/api/v1beta3/dynakube/oneagent_props.go#L145-L155

Reply:
You are right, nice catch 👍